contracts/staking/test/zrx_vault_backstop_test.ts

```diff
@@ -0,0 +1,50 @@
+import { ERC20Wrapper } from '@0x/contracts-asset-proxy';
+import { blockchainTests, describe, expect, increaseTimeAndMineBlockAsync } from '@0x/contracts-test-utils';
+import { StringRevertError } from '@0x/utils';
+import { LogWithDecodedArgs } from 'ethereum-types';
+
+import { ZrxVaultInCatastrophicFailureModeEventArgs } from '../src';
+
+import { deployAndConfigureContractsAsync, StakingApiWrapper } from './utils/api_wrapper';
+blockchainTests.resets('ZrxVaultBackstop', env => {
+    let stakingApiWrapper: StakingApiWrapper;
+    let authorizedAddress: string;
+    let notAuthorizedAddress: string;
+    before(async () => {
+        const accounts = await env.web3Wrapper.getAvailableAddressesAsync();
+        [authorizedAddress, notAuthorizedAddress] = accounts;
+        const erc20Wrapper = new ERC20Wrapper(env.provider, [], authorizedAddress);
+        stakingApiWrapper = await deployAndConfigureContractsAsync(env, authorizedAddress, erc20Wrapper);
+    });
+
+    describe('enterCatastrophicFailureIfProlongedReadOnlyMode', () => {
+        it('should revert if read-only mode is not set', async () => {
+            const expectedError = new StringRevertError('READ_ONLY_MODE_NOT_SET');
+            expect(
+                stakingApiWrapper.zrxVaultBackstopContract.enterCatastrophicFailureIfProlongedReadOnlyMode.awaitTransactionSuccessAsync(),
+            ).to.revertWith(expectedError);
+        });
+        it('should revert if read-only mode has been set for less than 40 days', async () => {
+            await stakingApiWrapper.stakingProxyContract.setReadOnlyMode.awaitTransactionSuccessAsync(true, {
+                from: authorizedAddress,
+            });
+            const expectedError = new StringRevertError('READ_ONLY_MODE_DURATION_TOO_SHORT');
+            expect(
+                stakingApiWrapper.zrxVaultBackstopContract.enterCatastrophicFailureIfProlongedReadOnlyMode.awaitTransactionSuccessAsync(),
+            ).to.revertWith(expectedError);
+        });
+        it('should enter catastophic failure mode if read-only mode has been set for 40 days', async () => {
```

Comment:
Optional Nit: A test that calls this function twice would be nice.

Reply:
We do test for that in the ZRXVault tests.

Reply:
That's why I made this optional. I think it's nice to make sure that this function is idempotent, but it would be somewhat redundant.